src/UhdmAst.cpp

```diff
@@ -1264,30 +1259,238 @@ AstVar* process_param_assign(vpiHandle obj_h, UhdmShared& shared) {
     return parameterp;
 }
 
+AstPackageImport* process_uhdm_import(vpiHandle obj_h, UhdmShared& shared) {
+    std::string objectName = get_object_name(obj_h);
+
+    AstPackage* packagep = nullptr;
+    auto it = shared.package_map.find(objectName);
+    if (it != shared.package_map.end()) { packagep = it->second; }
+    if (packagep != nullptr) {
+        std::string symbol_name;
+        vpiHandle imported_name = vpi_handle(vpiImport, obj_h);
+        if (imported_name) {
+            s_vpi_value val;
+            vpi_get_value(imported_name, &val);
+            symbol_name = val.value.str;
+        }
+        auto* package_importp
+            = new AstPackageImport(new FileLine("uhdm"), packagep, symbol_name);
+        shared.m_symp->importItem(packagep, symbol_name);
+        return package_importp;
+    }
+    return nullptr;
+}
+
+    AstMemberDType* process_typespec_member(vpiHandle obj_h, UhdmShared& shared) {
+    std::string objectName = get_object_name(obj_h);
+    AstNodeDType* typespec = nullptr;
+    auto typespec_h = vpi_handle(vpiTypespec, obj_h);
+    typespec = getDType(typespec_h, shared);
+    if (typespec != nullptr) {
+        auto* member = new AstMemberDType(new FileLine("uhdm"), objectName, VFlagChildDType(),
+                                          reinterpret_cast<AstNodeDType*>(typespec));
+        return member;
+    }
+    return nullptr;
+}
+
+AstNode* process_typespec(vpiHandle obj_h, UhdmShared& shared) {
+    std::string objectName = get_object_name(obj_h);
+    const unsigned int objectType = vpi_get(vpiType, obj_h);
+
+    auto file_name = vpi_get_str(vpiFile, obj_h);
+    const unsigned int currentLine = vpi_get(vpiLineNo, obj_h);
+    UINFO(6, "Object: " << objectName << " of type " << objectType << " ("
+                        << UHDM::VpiTypeName(obj_h) << ")"
+                        << " @ " << currentLine << " : " << (file_name != 0 ? file_name : "?")
+                        << std::endl);
```

Comment:
I didn't notice it. 
Do you have any idea how it can be changed? If I remove it, it won't be printed at all if we enter to that function not through `visit_object`.

Reply:
This was more useful back when `visit_object` was the only handling function, to identify the node being parsed. With other `process` functions and after #310 this is much easier.
You can leave this for debugging, just make sure this does not print the exact same content as `visit_object` - easiest way to do that would be to use `__func__` somewhere.